src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -116,42 +118,66 @@ protected void runTool(final JavaSparkContext ctx) {
         //TOOO: should this use getUnfilteredReads? getReads will apply default and command line filters
         final JavaRDD<GATKRead> initialReads = getReads();
 
-        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder(), getRecommendedNumReducers());
+        List<GATKRead> firstReads = initialReads.take(2);
```

Comment:
This seems like a kind of sketchy check.  It's very plausible that you'd just randomly have 2 unmapped reads at the start of your file.  

I think that if someone specifies an image file, then they are saying they want to align, and if they don't pass it in they don't want to align.

Reply:
@lbergelson I agree that this isn't reliable: I added it as a demonstration really. The problem is that specifying an image file is optional - it's inferred from the name of the reference since #3643.

So perhaps we do need an `--align` flag, although @tedsharpe doesn't like that... Is there a reliable way to determine if the reads have been aligned by looking at the contents?

Similarly if unmapped reads can pop up anywhere, then determining single or paired alignment is unreliable.

Any suggestions on how to move forward on this?